src/qinfer/domains.py

```diff
@@ -0,0 +1,534 @@
+        """
+        The numpy dtype of a single element of the domain.
+
+        :type: `np.dtype`
+        """
+        return np.dtype([('k', np.int, self.n_elements)])
+
+    @property
+    def n_members(self):
+        """
+        Returns the number of members in the domain if it 
+        `is_finite`, otherwise, returns `None`.
+
+        :type: ``int``
+        """
+        return int(binom(self.n_meas + self.n_elements -1, self.n_elements - 1))
+
+    @property
+    def example_point(self):
+        """
+        Returns any single point guaranteed to be in the domain, but 
+        no other guarantees; useful for testing purposes. 
+        This is given as a size 1 ``np.array`` of type ``dtype``.
+
+        :type: ``np.ndarray``
+        """
+        return np.array([([self.n_meas] + [0] * (self.n_elements-1))], dtype=self.dtype)
+
+    @property
+    def values(self):
+        """
+        Returns an `np.array` of type `self.dtype` containing 
+        some values from the domain.
+        For domains where ``is_finite`` is ``True``, all elements 
+        of the domain will be yielded exactly once.
+
+        :rtype: `np.ndarray`
+        """
+
+        # This code comes from Jared Goguen at http://stackoverflow.com/a/37712597/1082565
+        partition_array = np.empty((self.n_members, self.n_elements), dtype=int)
+        masks = np.identity(self.n_elements, dtype=int)
+        for i, c in enumerate(combinations_with_replacement(masks, self.n_meas)): 
+            partition_array[i,:] = sum(c)
+
+        # Convert to dtype before returning
+        return self.from_regular_array(partition_array)
+        
+    ## METHODS ##
+
+    def to_regular_array(self, A):
+        """
+        Converts from an array of type `self.dtype` to an array 
+        of type `int` with an additional index labeling the 
+        tuple indeces.
+
+        :param np.ndarray A: An `np.array` of type `self.dtype`.
+
+        :rtype: `np.ndarray`
+        """
+        # this could be a static method, but we choose to be consistent with 
+        # from_regular_array
+        return A.view((int, len(A.dtype.names))).reshape(A.shape + (-1,))
+
+    def from_regular_array(self, A):
+        """
+        Converts from an array of type `int` where the last index 
+        is assumed to have length `self.n_elements` to an array 
+        of type `self.d_type` with one fewer index.
+
+        :param np.ndarray A: An `np.array` of type `int`.
+
+        :rtype: `np.ndarray`
+        """
+        dims = A.shape[:-1]
+        return A.reshape((np.prod(dims),-1)).view(dtype=self.dtype).squeeze(-1).reshape(dims)
+
+    def in_domain(self, points):
+        """
+        Returns ``True`` if all of the given points are in the domain, 
+        ``False`` otherwise.
+
+        :param np.ndarray points: An `np.ndarray` of type `self.dtype`.
+
+        :rtype: `bool`
+        """
+        array_view = self.to_regular_array(points)
+        print(np.all(array_view>=0))
```

Comment:
If you must include a `print` call, which I don't usually suggest, be sure to to include `from __future__ import print_function`